src/main/java/io/cdap/plugin/http/common/pagination/BaseHttpPaginationIterator.java

```diff
@@ -122,8 +130,15 @@ protected BasePage getNextPage() throws IOException {
       case SUCCESS:
         break;
       case STOP:
-        throw new IllegalStateException(String.format("Fetching from url '%s' returned status code '%d' and body '%s'",
-                                                      nextPageUrl, httpStatusCode, response.getBody()));
+        String supportedDocUrl = "https://datatracker.ietf.org/doc/html/rfc7231#section-6";
+        String errorMessage = String.format("Retry failed! Unable to read new page and execute request. " +
+                        "Fetching from '%s' returned http error status code '%s'. For more details, see %s",
+                config.getUrl(), httpStatusCode, supportedDocUrl);
+        String errorReason = "Retry failed. Unable to read new page.";
```

Comment:
ErrorUtils.ActionErrorPair pair = ErrorUtils.getActionErrorByStatusCode(statusCode);
String errorReason = String.format("Unable to read new page: %s %s. %s", e.getStatusCode(), e.getStatusMessage(), pair.getCorrectiveAction());

This comment applies to whole PR.